mithril-common/src/cardano_block_scanner/interface.rs

```diff
@@ -55,18 +56,34 @@ pub trait BlockScanner: Sync + Send {
     ) -> StdResult<Box<dyn BlockStreamer>>;
 }
 
+/// [ChainScannedBlocks] allows to scan new blocks and handle rollbacks
+#[derive(Debug, PartialEq)]
+pub enum ChainScannedBlocks {
+    /// Roll forward on the chain to the next list of [ScannedBlock]
+    RollForwards(Vec<ScannedBlock>),
+    /// Roll backward on the chain to the previous [ChainPoint]
+    RollBackward(ChainPoint),
+}
+
 /// Trait that define how blocks are streamed from a Cardano database
 #[async_trait]
 pub trait BlockStreamer: Sync + Send {
     /// Stream the next available blocks
-    async fn poll_next(&mut self) -> StdResult<Option<Vec<ScannedBlock>>>;
+    async fn poll_next(&mut self) -> StdResult<Option<ChainScannedBlocks>>;
 
     /// Stream all the available blocks, may be very memory intensive
     async fn poll_all(&mut self) -> StdResult<Vec<ScannedBlock>> {
```

Comment:
We should probably change the return type of this function:
```suggestion
    async fn poll_all(&mut self) -> StdResult<Vec<ChainScannedBlocks>> {
```

Or should we remove this method from the trait as it used only for test purpose?

Reply:
We've removed it from the trait and created a new trait `BlockStreamerTestExtensions` dedicated to testing.
Commit: https://github.com/input-output-hk/mithril/pull/1716/commits/546a7d9c71ee387fdcb685db113ba30924d7d64a